components/ai_chat/resources/page/components/custom_label/index.tsx

```diff
@@ -0,0 +1,113 @@
+} {
+  switch (actionType) {
+      case mojom.ActionType.SUMMARIZE_SELECTED_TEXT:
+        return { category: undefined, item: getLocale('summarizeLabel')}
+      case mojom.ActionType.EXPLAIN:
+        return { category: undefined, item: getLocale('explainLabel')}
+      case mojom.ActionType.CREATE_TAGLINE:
+        return {
+          category: getLocale('createCategoryTitle'),
+          item: getLocale('taglineLabel')
+        }
+      case mojom.ActionType.CREATE_SOCIAL_MEDIA_COMMENT_SHORT:
+        return {
+          category: getLocale('socialMediaPostLabel'),
+          item: getLocale('socialMediaShortLabel')
+        }
+      case mojom.ActionType.CREATE_SOCIAL_MEDIA_COMMENT_LONG:
+        return {
+          category: getLocale('socialMediaPostLabel'),
+          item: getLocale('socialMediaLongLabel')
+        }
+      case mojom.ActionType.PARAPHRASE:
+        return {
+          category: getLocale('rewriteCategoryTitle'),
+          item: getLocale('paraphraseLabel')
+        }
+      case mojom.ActionType.IMPROVE:
+        return {
+          category: getLocale('rewriteCategoryTitle'),
+          item: getLocale('improveLabel')
+        }
+      case mojom.ActionType.SIMPLIFY:
+        return {
+          category: getLocale('rewriteCategoryTitle'),
+          item: getLocale('simplifyLabel')
+        }
+      case mojom.ActionType.ACADEMICIZE:
+        return {
+          category: getLocale('changeToneLabel'),
+          item: getLocale('academicizeLabel')
+        }
+      case mojom.ActionType.PROFESSIONALIZE:
+        return {
+          category: getLocale('changeToneLabel'),
+          item: getLocale('professionalizeLabel')
+        }
+      case mojom.ActionType.CASUALIZE:
+        return {
+          category: getLocale('changeToneLabel'),
+          item: getLocale('casualizeLabel')
+        }
+      case mojom.ActionType.FUNNY_TONE:
+        return {
+          category: getLocale('changeToneLabel'),
+          item: getLocale('funnyToneLabel')
+        }
+      case mojom.ActionType.SHORTEN:
+        return {
+          category: getLocale('changeLengthLabel'),
+          item: getLocale('shortenLabel')
+        }
+      case mojom.ActionType.EXPAND:
+        return {
+          category: getLocale('changeLengthLabel'),
+          item: getLocale('expandLabel')
+        }
+      default:
+        return { category: undefined, item: undefined }
+  }
+}
+
+function CustomLabel (props: CustomLabelProps) {
+  const { category, item } = getCategoryAndItem(props.actionType)
+
+  if (!category) {
+    return (
+      <div className={styles.customLabel}>
+        <span className={styles.text}>{item}</span>
+      </div>
+    )
+  }
+
+  if (category && item) {
+    return (
+      <div className={styles.customLabel}>
+        <span className={styles.text}>{category}</span>
+        <Icon name='carat-right' />
+        <span className={styles.text}>{item}</span>
+      </div>
+    )
+  }
+
+  return null
```

Comment:
As discussed in the DM, we don't expect to see those cases when selectedText is not null.
We decided to render like below when these happens to avoid returning null and also easier to catch bug if this do ever happen in the future.
<img width="718" alt="Screenshot 2024-02-23 at 11 27 44 AM" src="https://github.com/brave/brave-core/assets/4730197/64768f63-4856-4cd1-b371-a075c3e68ce9">
https://github.com/brave/brave-core/pull/22090/commits/8bcae2d866c79ffea8feb8f0e5ba5e7462a8a660

Reply:
to summarize, if `ActionTypeLabel` does not render, that means `selectedText` at the call site is null. if it renders without text, then `actionType` is in an undefined state that needs resolution.